### PR TITLE
[IMP] runbot_travis2docker: Adding weblate integration over runbot_travis2docker

### PR DESCRIPTION
**Pull request by JesusZapata**
Related https://github.com/OCA/maintainer-quality-tools/pull/439
WIP

### CODE REVIEW

File: runbot_travis2docker/models/runbot_build.py [hunk 1]
```diff
@@ -111,6 +112,18 @@ def job_20_test_all(self, cr, uid, build, lock_path, log_path):
         travis_branch = build._get_closest_branch_name(
             build.repo_id.id
         )[1].split('/')[-1]
+        wl_cmd_env = []
+        if build.uses_weblate and 'refs/pull' not in build.branch_id.name:
+            wl_cmd_env = [
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: use `wl_cmd_env.extend([...])`

File: runbot_travis2docker/models/runbot_build.py [hunk 2]
```diff
@@ -111,6 +112,18 @@ def job_20_test_all(self, cr, uid, build, lock_path, log_path):
...
+                '-e', 'WEBLATE=1',
+                '-e', ('WEBLATE_TOKEN=%s' %
+                       build.branch_id.repo_id.weblate_token),
+                '-e', ('WEBLATE_HOST=%s' %
+                       build.branch_id.repo_id.weblate_url)
+            ]
+            if build.branch_id.repo_id.token:
+                wl_cmd_env = wl_cmd_env + ['-e', 'GITHUB_TOKEN=%s' %
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: use `wl_cmd_env.extend([...])`

File: runbot_travis2docker/models/runbot_build.py [hunk 3]
```diff
@@ -123,9 +136,9 @@ def job_20_test_all(self, cr, uid, build, lock_path, log_path):
                 not build.repo_id.travis2docker_test_disable),
             '-p', '%d:%d' % (build.port, 8069),
             '-p', '%d:%d' % (build.port + 1, 22),
-            '--name=' + build.docker_container,
-            '-t', build.docker_image,
-        ] + pr_cmd_env
+        ] + pr_cmd_env + wl_cmd_env
+        cmd = cmd + ['--name=' + build.docker_container,
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: Use `cmd.extend([...])`

File: runbot_travis2docker/models/runbot_repo.py [hunk 4]
```diff
@@ -3,11 +3,32 @@
...
+from openerp import _, fields, models, api
+from openerp.exceptions import ValidationError
 
 
 class RunbotRepo(models.Model):
     _inherit = "runbot.repo"
 
     is_travis2docker_build = fields.Boolean('Travis to docker build')
     travis2docker_test_disable = fields.Boolean('Test Disable?')
+    weblate_url = fields.Char(default="https://weblate.vauxoo.com/api")
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_repo.py]: Use oca url

File: runbot_travis2docker/models/runbot_build.py [hunk 5]
```diff
@@ -111,6 +112,18 @@ def job_20_test_all(self, cr, uid, build, lock_path, log_path):
...
+            wl_cmd_env += [
+                '-e', 'WEBLATE=1',
+                '-e', ('WEBLATE_TOKEN=%s' %
+                       build.branch_id.repo_id.weblate_token),
+                '-e', ('WEBLATE_HOST=%s' %
+                       build.branch_id.repo_id.weblate_url)
+            ]
+            if build.branch_id.repo_id.token:
+                wl_cmd_env += [
+                    '-e', 'GITHUB_TOKEN=%s' % build.branch_id.repo_id.token]
```
Comment (moylop260) [inline runbot_travis2docker/models/runbot_build.py]: Use GITLAB_TOKEN too if there is field `uses_gitlab` and is enabled

File: runbot_travis2docker/models/runbot_repo.py (hunk 4 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_repo.py]: Done

File: runbot_travis2docker/models/runbot_build.py (hunk 5 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_build.py]: I add `GITLAB_TOKEN` container with the `build.branch_id.repo_id.token` value

File: runbot_travis2docker/models/runbot_build.py (hunk 3 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_build.py]: Done

File: runbot_travis2docker/models/runbot_build.py (hunk 2 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_build.py]: Done

File: runbot_travis2docker/models/runbot_build.py (hunk 1 above)
Comment (JesusZapata) [reply runbot_travis2docker/models/runbot_build.py]: Done

**Comment by JesusZapata** [comment]
@moylop260 
In this PR contains another PR's
https://github.com/Vauxoo/runbot-addons/pull/121
https://github.com/Vauxoo/runbot-addons/pull/120
https://github.com/Vauxoo/runbot-addons/pull/119
https://github.com/Vauxoo/runbot-addons/pull/118
https://github.com/Vauxoo/runbot-addons/pull/114
https://github.com/Vauxoo/runbot-addons/pull/110
https://github.com/Vauxoo/runbot-addons/pull/108

**Comment by JesusZapata** [comment]
@moylop260 
In this PR contains another PR
https://github.com/Vauxoo/runbot-addons/pull/122

**Comment by JesusZapata** [comment]
@moylop260 
In this PR contains another PR
https://github.com/Vauxoo/runbot-addons/pull/123
https://github.com/Vauxoo/runbot-addons/pull/125
https://github.com/Vauxoo/runbot-addons/pull/126
https://github.com/Vauxoo/runbot-addons/pull/127
https://github.com/Vauxoo/runbot-addons/pull/128

**Review by max3903 — APPROVED** [review]
:+1: LGTM no test

**Review by pedrobaeza — APPROVED** [review]
Code review. Let's merge and see using it if we find some problems.

**Pull request merged**